### PR TITLE
Fixed DNS resolution on android

### PR DESCRIPTION
**Pull request by justinboswell**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: source/posix/host_resolver.c [hunk 1]
```diff
@@ -41,13 +41,18 @@ int aws_default_dns_resolve(
...
+#ifdef ANDROID
+    int err_code = getaddrinfo(hostname_cstr, NULL, NULL, &result);
+#else
     struct addrinfo hints;
     AWS_ZERO_STRUCT(hints);
     hints.ai_family = AF_UNSPEC;
     hints.ai_socktype = SOCK_STREAM;
     hints.ai_flags = AI_ALL | AI_V4MAPPED;
 
     int err_code = getaddrinfo(hostname_cstr, NULL, &hints, &result);
```
Comment (bretambrose) [inline source/posix/host_resolver.c:54]: Do we still getting a union of v4 and v6 in the results?

Comment (justinboswell) [reply source/posix/host_resolver.c:54]: If you're asking if we support it, yes we do.

Comment (bretambrose) [reply source/posix/host_resolver.c:54]: To clarify was just wondering it that's what we get back without hints.

Comment (justinboswell) [reply source/posix/host_resolver.c:54]: Yes. It boils down to what the interfaces available support, and it's a union between them.

**Pull request merged**